### PR TITLE
feat: add unit tests for useMutationObserver

### PR DESCRIPTION
**Pull request by mrdulin**
1. Added unit tests for the `useMutationObserver` hook.
2. Removed unused variable.

### CODE REVIEW

**Comment by imbhargav5** [comment]
Sheesh! Yet again, awesome work!

File: src/__tests__/useMutationObserver.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    }
+    render(<TestComp />);
+    const button = screen.getByText(/start to play game!/i);
+    fireEvent.click(button);
+    await waitFor(() => {
+      expect(calls).toBe(2);
+    });
+  });
+
+  test("should stop the MutationObserver instance from receiving further notifications", async () => {
```
Comment (imbhargav5) [inline src/__tests__/useMutationObserver.spec.tsx:116]: We could name this test slightly better, but this doesn't need to happen in this PR. Looks great overall!

**Review by imbhargav5 — APPROVED** [review]

File: src/__tests__/useMutationObserver.spec.tsx (hunk 1 above)
Comment (mrdulin) [reply src/__tests__/useMutationObserver.spec.tsx:116]: My English is poor. Some statements may not conform to common English usage. Help me fix it. Thanks.

Comment (imbhargav5) [reply src/__tests__/useMutationObserver.spec.tsx:116]: Don't worry about it! English is not the most important thing here.

**Pull request merged**